_publications/jors/paper.tex

```diff
@@ -0,0 +1,457 @@
+communities with limited funding and staffing.  The framework relies
+on the API, WebSocket and webhook infrastructure that underpins the
+modern web and allows external services to have full-duplex
+communication between servers and browsers. In particular, PFHub
+relies on GitHub's well maintained API and webhook functionality for
+external services (such as Travis CI~\cite{travis} and
+Staticman~\cite{staticman}).
+
+The workflow for uploading benchmark results relies on third party
+tools using the following steps, illustrated in
+Figure~\ref{fig:pfhub_website}.
+\begin{enumerate}
+  \item The users are first required to upload simulation outputs to
+    an archival resource (\emph{e.g.}, Figshare\footnote{Certain
+      commercial equipment, instruments, or materials (or suppliers,
+      or software, ...) are identified in this paper to foster
+      understanding. Such identification does not imply recommendation
+      or endorsement by the National Institute of Standards and
+      Technology, nor does it imply that the materials or equipment
+      identified are necessarily the best available for the
+      purpose.\label{disclaimer}}~\cite{figshare}) configured with
+    permissive cross-origin resource sharing (CORS).
+  \item The metadata summarizing each simulation is entered into a
+    form on the website, including relevant details such as memory
+    usage, run time and links to the data archived in the first
+    step~\ref{fig:pfhub_upload}.
+  \item Upon submission, the Staticman app~\cite{staticman} submits
+    the entered metadata as a pull request against the PFHub
+    repository hosted by GitHub. The metadata is stored in a YAML file
+    with a unique path in the repository, see~\cite{upload_data} for
+    an example.
+  \item Travis CI~\cite{travis} performs linting on the submission and
+    then launches a temporary version of the proposed website using
+    Surge~\cite{surge}. The PFHub admins can then examine the new
+    submission and further changes can be made if necessary.
+  \item Once review has been completed to the satisfaction of both the
+    uploading scientist and the website maintainers, the pull request
+    is merged and served to the World Wide Web using a hosting service
+    compatible with GitHub Pages.
+\end{enumerate}
+A combination of Jekyll templates and CoffeeScript are used to access
+and download the data links in the submitted YAML files and then
+display the data in interactive plots on the website. CoffeeScript is
+a higher level language than JavaScript and, thus, more readable
+whilst also allowing programming in a more functional manner which
+makes data manipulation pipelines both more succinct and easier to
+understand. The interactive plots are displayed using the Plotly
+JavaScript Graphing Library~\cite{plotly} as it provides a
+programmable interface and requires minimal
+configuration~\ref{fig:pfhub_results}, see~\cite{display_data} for
```

Comment:
see above comment about clarifying this is supposed to point to a figure

Reply:
done